diff --git a/colors/gruvbox.vim b/colors/gruvbox.vim
index 66246fb..5ae492f 100644
--- a/colors/gruvbox.vim
+++ b/colors/gruvbox.vim
@@ -86,10 +86,10 @@ let s:is_dark=(&background == 'dark')
 let s:gb = {}
 
 " fill it with absolute colors
-let s:gb.dark0_hard  = ['#1d2021', 234]     " 29-32-33
+let s:gb.dark0_hard  = ['#000000', 234]     " 29-32-33
 let s:gb.dark0       = ['#282828', 235]     " 40-40-40
 let s:gb.dark0_soft  = ['#32302f', 236]     " 50-48-47
-let s:gb.dark1       = ['#3c3836', 237]     " 60-56-54
+let s:gb.dark1       = ['#222222', 237]     " 60-56-54
 let s:gb.dark2       = ['#504945', 239]     " 80-73-69
 let s:gb.dark3       = ['#665c54', 241]     " 102-92-84
 let s:gb.dark4       = ['#7c6f64', 243]     " 124-111-100
@@ -595,7 +595,7 @@ hi! link Keyword GruvboxRed
 " Variable name
 hi! link Identifier GruvboxBlue
 " Function name
-hi! link Function GruvboxGreenBold
+hi! link Function GruvboxOrange
 
 " Generic preprocessor
 hi! link PreProc GruvboxAqua
@@ -916,6 +916,10 @@ call s:HL('CocHintHighlight', s:none, s:none, s:undercurl, s:blue)
 " }}}
 
 " Filetype specific -----------------------------------------------------------
+" C++: {{{
+hi! link TSVariable GruvboxBlue
+hi! link TSFunction GruvboxOrange
+" }}}
 " Diff: {{{
 
 hi! link diffAdded GruvboxGreen
